tmt/steps/report/reportportal.py

```diff
@@ -15,12 +15,12 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
     url: Optional[str] = field(
         option="--url",
         metavar="URL",
-        default=os.environ.get("TMT_REPORT_REPORTPORTAL_URL"),
+        default=None,
         help="The URL of the ReportPortal instance where the data should be sent to.")
     token: Optional[str] = field(
         option="--token",
         metavar="TOKEN",
-        default=os.environ.get("TMT_REPORT_REPORTPORTAL_TOKEN"),
+        default=None,
```

Comment:
IIUIC, the `os.environ` is necessary. See the note at https://tmt.readthedocs.io/en/stable/overview.html#plugin-variables: tmt will extract field value from environment variable only when the step and plugin are mentioned on the command line. This means, in the general case of TF, with the plugin listed in fmf files only, the environment variables would be ignored and have no effect. `os.environ` is a workaround: when executed, the `default` is filled with whatever is in the environment, tricking tmt into honoring the environment after all.

Reply:
I noticed you moved `os.environ` use into the code - I'd recommend against it, as it splits field value initialization into several places. One day, environment variables would be honored even without listing the plugin on the command line, `os.environ` trick wouldn't be necessary, and hunting down all those suddenly useless `os.environ.get()` calls would be easier should they happen in step data classes only.

Reply:
I followed the implementation of polarion plugin. My intention was mostly to change it into TMT_PLUGIN_REPORT_REPORTPORTAL_<option> form (instead of TMT_REPORT_REPORTPORTAL_<token/url>) and have it uniform.